src/phone-number-input/types.ts

```diff
@@ -0,0 +1,9 @@
+import { FormInputProps } from "src/form/types";
+import { InputProps } from "../input/types";
+import { ListAddon } from "src/input-group";
+
+export interface PhoneNumberInputProps<T, V>
+    extends InputProps,
+        FormInputProps {
+    countrySelector?: ListAddon<T, V>;
```

Comment:
I might have forgotten to mention previously. I believe this component is meant to provide an exhaustive list required. I think we need not expose the ListAddon props. I also realise it might not be a good idea to plainly inherit the InputProps. As such here are my proposal for the props:

```
export interface PhoneNumberInputValue {
    country?: string | undefined;
    number?: string | undefined;
}
export interface PhoneNumberInputProps
    extends React.HTMLAttributes<HTMLInputElement> {
    value?: PhoneNumberInputValue | undefined;
    onChange?: (value: PhoneNumberInputValue) => void | undefined;
    error?: boolean | undefined;
    /** Allows clearing of input. Specifying true will render a cross icon button */
    allowClear?: boolean | undefined;
    onClear?: () => void | undefined;
    "data-testid"?: string | undefined;

    // Dropdown option props
    optionSearchPlaceholder?: string | undefined;
    /** Specifying will render a search bar in the dropdown */
    enableSearch?: boolean | undefined;
    /** Called when the country code options is dismissed */
    onHideOptions?: () => void | undefined;
     /** Called when the country code options is shown */
    onShowOptions?: () => void | undefined;

    // Input props
    disabled?: boolean | undefined;
    name?: string | undefined;
    pattern?: string | undefined;
    placeholder?: string | undefined;
    readOnly?: boolean | undefined;
}
```

Your valueExtractor and listExtractor and all should reside within this component instead of having to pass from outside.

Reply:
The custom `onChange` will be fired whenever the user enters/clear something in the input or selects an option in the dropdown

Reply:
I also feel can just standardise the display to all have the secondary label.